pkg/scheduler/core/extender_test.go

```diff
@@ -112,20 +113,80 @@ type FakeExtender struct {
 	filteredNodes    []*v1.Node
 }
 
+// selectVictimsOnNodeByExtender checks the given nodes->pods map with predicates on extender's side.
+// Returns:
+// 1. More victim pods (if any) amended by preemption phase of extender.
+// 2. Number of violating victim (used to calculate PDB).
+// 3. Fits or not after preemption phase on extender's side.
+func (f *FakeExtender) selectVictimsOnNodeByExtender(
+	pod *v1.Pod,
+	node *v1.Node,
+	originVictims *schedulerapi.Victims,
+	pdbs []*policy.PodDisruptionBudget,
+) ([]*v1.Pod, int, bool) {
+	if fits, _ := f.runPredicate(pod, node); !fits {
+		return nil, 0, false
+	}
+
+	return []*v1.Pod{}, 0, true
+}
+
+func (f *FakeExtender) SupportsPreemption() bool {
+	return true
+}
+
+func (f *FakeExtender) ProcessPreemption(
+	pod *v1.Pod,
+	nodeToVictims map[string]*schedulerapi.Victims,
+	nodeNameToInfo map[string]*schedulercache.NodeInfo,
+	pdbs []*policy.PodDisruptionBudget,
+) (schedulerapi.ExtenderPreemptionResult, error) {
+	nodeToVictimsCopy := map[string]*schedulerapi.Victims{}
+	// We don't want to change the original nodeToVictims
+	for k, v := range nodeToVictims {
+		nodeToVictimsCopy[k] = v
+	}
+
+	for nodeName, victims := range nodeToVictimsCopy {
+		// Try to do preemption on extender side.
+		extenderVictimPods, extendernPDBViolations, fits := f.selectVictimsOnNodeByExtender(pod, nodeNameToInfo[nodeName].Node(), victims, pdbs)
+		// If it's unfit after extender's preemption, this node is unresolvable by preemption overall,
+		// let's remove it from potential preemption nodes.
+		if !fits {
+			delete(nodeToVictimsCopy, nodeName)
+		} else {
+			// Append new victims to original victims
```

Comment:
Is the protocol here that extender may pick additional victims and only returns those?

Reply:
Yes, we discussed this before and decide to make this assumption.